### PR TITLE
[MERGE][IMP] web: pager scss revamp

### PR DESCRIPTION
**Pull request by Brieuc-brd**
This PR simplifies the SCSS of the Pager component.

After the revamp of the pager, the Legacy Pager was broken because it
used the same SCSS file. This commit adds some BS classes on the Legacy
Pager to keep consistency between both versions while waiting for the
removal of the legacy one.

task-2733437

### CODE REVIEW

**Comment by Brieuc-brd** [comment]
Hello @stefanorigano,
New component done, only one CSS rule! 💪 
Do you think we can go further?

**Review by stefanorigano — CHANGES_REQUESTED** [review]
It's broken in the form view, I didn't test further  :/
image

note: commit messages titles are always lowercase

**Comment by Brieuc-brd** [comment]
> 

@stefanorigano  It's probably the legacy version that you've seen in the form view.
According to MCM, the new Pager is displayed on views which are already rewritten in 15.0 or 15.1, like Gantt, Pivot, Cohort and Map.

**Comment by stefanorigano** [comment]
> It's probably the legacy version that you've seen in the form view (it still uses the Font Awesome icons).

Well, this cannot be merged anyway :/ Check if you can apply your changes without affecting the legacy one.
If it's not possible consider adding SCSS in the legacy file, but:
- the OWL version must be perfect even without this code (legacy files will be removed)
- add notes in the legacy SCSS explaining why you did that

**Comment by Brieuc-brd** [comment]
> 
@stefanorigano 👋 
The legacy Pager is using the SCSS file of the OWL version. To keep consistency, I created a new commit which adding BS classes on the Legacy Pager and explaining why in the commit message.

**Comment by stefanorigano** [comment]
@Brieuc-brd , you can remove the draft state ;)

@aab-odoo , ready for tech review. Please note that BRD applied changes on the legacy component too  https://github.com/odoo/odoo/pull/82866/commits/883d05b6b621fcbeaeb94c1360b3dc42908053b1. Nothing critical IMO, but just for you be aware ;)

cc. @adr-odoo

File: addons/mail/static/src/components/messaging_menu/messaging_menu.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <t t-name="mail.MessagingMenu" owl="1">
         <div class="o_MessagingMenu dropdown" t-att-class="{ 'show': messagingMenu ? messagingMenu.isOpen : false, 'o-mobile': messaging ? messaging.device.isMobile : false }" t-attf-class="{{ className }}" t-ref="root">
             <a class="o_MessagingMenu_toggler dropdown-toggle o-no-caret o-dropdown--narrow" t-att-class="{ 'o-no-notification': messagingMenu ? !messagingMenu.counter : false }" href="#" title="Conversations" role="button" t-att-aria-expanded="messagingMenu and messagingMenu.isOpen ? 'true' : 'false'" aria-haspopup="true" t-on-click="_onClickToggler">
-                <i class="o_MessagingMenu_icon fa fa-comments" role="img" aria-label="Messages"/>
+                <i class="o_MessagingMenu_icon oi oi-forum" role="img" aria-label="Messages"/>
```
Comment (seb-odoo) [inline addons/mail/static/src/components/messaging_menu/messaging_menu.xml:7]: Change looks unrelated to pager, wrong PR split?
(but ok for the change of icon in general)

Comment (stefanorigano) [reply addons/mail/static/src/components/messaging_menu/messaging_menu.xml:7]: @Brieuc-brd has cherry-pick my commit with the new icons.
Icons will be  be merged either here or in the navbar branch https://github.com/odoo/odoo/pull/82434 (that PR it's taking longer than expected)

**Comment by adr-odoo** [comment]
Can I ask you to rebase your branch and check icons?

**Comment by Brieuc-brd** [comment]
> Can I ask you to rebase your branch and check icons?

Hey @adr-odoo 👋 
The branch is ready.

File: addons/web/static/src/core/pager/pager.scss [hunk 2]
```diff
@@ -1,24 +1,3 @@
...
-    
-    input.o_pager_value {
-        text-align: right;
-        width: 60px;
-        &:focus {
-            outline: none;
-        }
-    }
+.o_pager .o_pager_counter {
+    min-width: $o-statbutton-height;
```
Comment (adr-odoo) [inline addons/web/static/src/core/pager/pager.scss:2]: Seems not necessary as flex-basis is auto by default.

File: addons/web/static/src/core/pager/pager.xml [hunk 3]
```diff
@@ -2,13 +2,13 @@
 <templates xml:space="preserve">
 
     <t t-name="web.Pager" owl="1">
-        <nav class="o_pager" aria-label="Pager">
+        <nav class="o_pager align-items-center d-flex" aria-label="Pager">
             <span class="o_pager_counter mr-2" t-on-click.stop="">
```
Comment (adr-odoo) [inline addons/web/static/src/core/pager/pager.xml:6]: Can we remove mr-2 and use gap instead?

Comment (Brieuc-brd) [reply addons/web/static/src/core/pager/pager.xml:6]: Unfortunately BTS 4 does not support `gap` but this margin could be replaced by `.gap-2` after the migration to BTS5.

Comment (adr-odoo) [reply addons/web/static/src/core/pager/pager.xml:6]: Yes I know and it's why we use gap in scss to be able to easily find them to then modify once again the xml file with the correct BT5 class... So, as it's not the first time we need this class, I propose to backport it to do the job properly.

**Review by adr-odoo — APPROVED** [review]
@robodoo rebase-merge r+
